java/ql/test/stubs/spring-ldap-2.3.2/org/springframework/ldap/filter/HardcodedFilter.java

```diff
@@ -0,0 +1,6 @@
+package org.springframework.ldap.filter;
+
+public class HardcodedFilter implements Filter {
+  public HardcodedFilter(String filter) { }
+  public StringBuffer encode(StringBuffer buff) { return buff; }
+}
```

Comment:
The reason for the first negative (`sBadFilterToString` argument not detected) is the lack of `toString` method definition in `HardcodedFilter` class.
In the query, taint is propagated if `toString` method is invoked on tainted `Filter` (i.e. any class implementing `Filter` interface). Because the `toString` definition is missing here, the one from `Object` is invoked and it's not caught by the query.
Adding `toString` method returning `null` fixes the problem in my environment.